jerry-core/ecma/builtin-objects/ecma-builtin-promise.c

```diff
@@ -38,6 +41,621 @@
+
+  /* 8. */
+  ecma_string_t *str_index = ecma_new_ecma_string_from_uint32 ((uint32_t) ecma_get_integer_from_value (index_val));
+  ecma_op_object_put (ecma_get_object_from_value (value_array),
+                      str_index,
+                      argv[0],
+                      false);
+  ecma_deref_ecma_string (str_index);
+
+  /* 9-10. */
+  if (ecma_builtin_promise_remaining_inc_or_dec (remaining, false) == 0)
+  {
+    /* String '1' indicates [[Resolve]] of capability. */
+    ecma_value_t resolve = ecma_op_object_get (ecma_get_object_from_value (capability),
+                                               str_1);
+    ret = ecma_op_function_call (ecma_get_object_from_value (resolve),
+                                 ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED),
+                                 &value_array,
+                                 1);
+    ecma_free_value (resolve);
+  }
+
+  ecma_free_value (remaining);
+  ecma_free_value (capability);
+  ecma_free_value (value_array);
+  ecma_fast_free_value (index_val);
+  ecma_fast_free_value (already_called);
+  ecma_deref_ecma_string (str_0);
+  ecma_deref_ecma_string (str_1);
+  ecma_deref_ecma_string (str_2);
+  ecma_deref_ecma_string (str_3);
+  ecma_deref_ecma_string (str_4);
+
+  return ret;
+} /* ecma_builtin_promise_all_handler */
+
+/**
+ * Runtime Semantics: PerformPromiseAll.
+ *
+ * See also:
+ *         ES2015 25.4.4.1.1
+ *
+ * @return ecma value of the new promise.
+ *         Returned value must be freed with ecma_free_value.
+ */
+inline static ecma_value_t
+ecma_builtin_promise_do_all (ecma_value_t array, /**< the array for all */
+                             ecma_value_t capability, /**< PromiseCapability record */
+                             ecma_value_t ctor) /**< the caller of Promise.race */
+{
+
+  JERRY_ASSERT (ecma_is_value_object (capability)
+                && ecma_is_value_object (array)
+                && ecma_is_value_object (ctor));
+  JERRY_ASSERT (ecma_get_object_builtin_id (ecma_get_object_from_value (ctor)) == ECMA_BUILTIN_ID_PROMISE);
+  JERRY_ASSERT (ecma_get_object_type (ecma_get_object_from_value (array)) == ECMA_OBJECT_TYPE_ARRAY);
+
+  ecma_value_t ret = ecma_make_simple_value (ECMA_SIMPLE_VALUE_EMPTY);
+  ecma_string_t *magic_string_length_p = ecma_new_ecma_length_string ();
+  ecma_object_t *array_p = ecma_get_object_from_value (array);
+  ecma_value_t len_value = ecma_op_object_get (array_p, magic_string_length_p);
+  ecma_deref_ecma_string (magic_string_length_p);
+  ecma_length_t len = (ecma_length_t) ecma_get_integer_from_value (len_value);
+  ecma_fast_free_value (len_value);
+  ecma_string_t *str_0 = ecma_new_ecma_string_from_uint32 (0);
+  ecma_string_t *str_1 = ecma_new_ecma_string_from_uint32 (1);
+  ecma_string_t *str_2 = ecma_new_ecma_string_from_uint32 (2);
+  ecma_string_t *str_3 = ecma_new_ecma_string_from_uint32 (3);
+  ecma_string_t *str_4 = ecma_new_ecma_string_from_uint32 (4);
```

Comment:
Updated. You're right, since they are uint32-string, only the uint32 numbers are stored in the property, the total memory will not affected. 
Performance will be affected, because of the more `ecma_new_ecma_string_from_uint32` and `ecma_deref_ecma_string`, but only slightly effect.